atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/UniquenessChecker.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright 2016 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.jepsen;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.ImmutableList;
+import com.palantir.atlasdb.jepsen.events.Checker;
+import com.palantir.atlasdb.jepsen.events.Event;
+import com.palantir.atlasdb.jepsen.events.EventVisitor;
+import com.palantir.atlasdb.jepsen.events.FailEvent;
+import com.palantir.atlasdb.jepsen.events.InfoEvent;
+import com.palantir.atlasdb.jepsen.events.InvokeEvent;
+import com.palantir.atlasdb.jepsen.events.OkEvent;
+
+public class UniquenessChecker implements Checker {
+
+    private static class Visitor implements EventVisitor {
+        private final List<Event> errors = new ArrayList<>();
+        private boolean valid = true;
+        private final Map<Long, OkEvent> valuesAlreadySeen = new HashMap<>();
+
+        @Override
+        public void visit(InfoEvent event) {
+        }
+
+        @Override
+        public void visit(InvokeEvent event) {
+        }
+
+        @Override
+        public void visit(OkEvent event) {
+            long value = event.value();
+
+            if (valuesAlreadySeen.containsKey(value)) {
+                OkEvent previousEvent = valuesAlreadySeen.get(value);
+                valid = false;
+                errors.add(previousEvent);
+                errors.add(event);
+            }
+
+            valuesAlreadySeen.put(value, event);
+        }
+
+        @Override
+        public void visit(FailEvent event) {
+        }
+
+        public boolean valid() {
+            return valid;
+        }
+
+        public List<Event> errors() {
+            return ImmutableList.copyOf(errors);
+        }
+    }
+
+    @Override
+    public CheckerResult check(List<Event> events) {
```

Comment:
(To add to this, hope I'm not asking too much but it would be nice to modify `MonotonicChecker` to conform to this as well; generally from what I've seen in our codebase private static classes are at the end)